keras_cv/bounding_box/non_max_suppression.py

```diff
@@ -147,8 +147,8 @@ def _encode_to_ragged(self, nmsed_boxes, images):
             )
 
             # iterate through the boxes and append it to TensorArray
-            for i in range(nmsed_boxes.valid_detections[i]):
-                boxes = boxes.write(boxes.size(), boxes_recombined[i])
+            for j in range(nmsed_boxes.valid_detections[i]):
+                boxes = boxes.write(boxes.size(), boxes_recombined[j])
```

Comment:
any reason to keep using boxes.size() here now?

Reply:
This is because j will be reset for every batch, we need an append function

Reply:
Ohhh I see, I'm actually surprised that this works in graph mode.  What a wonderful trick, thanks for showing me this.